src/components/Deck/index.js

```diff
@@ -36,7 +37,9 @@ export default React.memo(function Deck(props) {
                 `Deck__card--${card.type}`,
                 card.rarity === 'legendary' && `Deck__card--legendary`,
                 highlightedCards.length > 0 &&
-                  !highlightedCards.includes(card.id) &&
+                  !highlightedCards.find(highlightedCard =>
+                    areCardsEqual(highlightedCard, card)
+                  ) &&
```

Comment:
I would probably use currying for the `areCardsEqual` helper:
```js
const areCardsEqual = a => b => { … }
```
```js
!highlightedCards.find(areCardsEqual(card))
```
Maybe we even name it `isCard` or something to make it look like `cards.find(isCard(card))`.